client/src/App.tsx

```diff
@@ -17,35 +17,47 @@ import RequestManagementContainer from "./components/RequestManagementContainer"
 export interface OwnProps {}
 
 interface StateProps {
-    a: number
-    user: User|null
+    a: number;
+    user: User|null;
 }
 
 type Props = StateProps & OwnProps;
 
 class App extends Component<Props, {}> {
-    async checkAuth(): Promise<User> {
-        return new Promise(resolve => setTimeout(() => resolve({
-            "uuid": "abcdedf-afdhkasdf-adfsk",
-            "name": "Evan Strat",
-            "isAdmin": true
-        }), 5000));
-    }
-
-    async componentWillMount(): Promise<void> {
-        const user: User = await this.checkAuth();
+    public async componentWillMount(): Promise<void> {
+        const userRequest = await fetch("/api", {
+            credentials: "include",
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+                query: `
+                        query {
+                          user {
+                            uuid
+                            name
+                            admin
+                          }
+                        }
+                    `
+            }),
+        });
+        const json = await userRequest.json();
+        const user = json.data.user;
```

Comment:
non-null checks would be good

Reply:
Done, although I want to refactor the fetch out of this component and into the action (or wherever the proper place is)